coffee/shepherd.coffee

```diff
@@ -117,6 +117,9 @@ class Step extends Evented
     @tether = new Tether extend(tetherOpts, @options.tetherOptions)
 
   show: =>
+      @options.beforeShowPromise?()?.then(=> @_show()) ? @_show()
```

Comment:
I tested on my end, and it works as we'd expect. And yes, it removes the anonymous wrapper, but it's unnecessary because `_show()` is a bound function (because of the fat arrow).